sycl/include/CL/sycl/detail/kernel_info.hpp

```diff
@@ -28,7 +28,7 @@ template <info::kernel Param> struct get_kernel_info_cl<string_class, Param> {
     if (ResultSize == 0) {
       return "";
     }
-    string_class Result(ResultSize, ' ');
+    string_class Result(ResultSize - 1, ' ');
     // TODO catch an exception and put it to list of asynchronous exceptions
     CHECK_OCL_CODE(clGetKernelInfo(ClKernel, cl_kernel_info(Param), ResultSize,
                                    &Result[0], nullptr));
```

Comment:
Since I am reading this, I am not sure it is correct in C++ to write to some `std::string` internal.
Ah, actually it seems correct since C++11 because there is no longer copy-on-write possible implementation...
What about using on purpose `Result.data()` instead of `&Result[0]`?
But per I wonder why all this runtime is based on the old C OpenCL API instead of the new C++ `cl2.hpp` one at least, by the way... :-(

Reply:
I made this logic without hacks.